src/charm.py

```diff
@@ -325,5 +353,29 @@ def _push_ca_cert_to_container(self, ca_certificate: str):
             f.write(ca_certificate)
 
 
+def parse_time_string(time_str: str) -> timedelta:
+    """Parse a given time string.
+
+    It must be a number followed by either an
+    s for seconds, h for hours, d for days or y for years.
```

Comment:
Would it make sense to support combinations of these?  i.e. `1y 1m`

It could always be added later without breaking compatibility, so not necessarily something we need to do now.

Reply:
I don't think many people would use it that way. I imagine the most common values to be 10s for testing, 1y for certificates, 3/5/10y for CA's. I just can't imagine someone coming to us to say "hey we would really like a certificate with a 1year+1day deadline.